yt_dlp/extractor/nebula.py

```diff
@@ -195,44 +128,156 @@ def _real_initialize(self):
         if nebula_cookie:
             self.to_screen('Authenticating to Nebula with token from cookie jar')
             nebula_cookie_value = compat_urllib_parse_unquote(nebula_cookie.value)
-            self._nebula_token = self._parse_json(nebula_cookie_value, None).get('apiToken')
+            self._nebula_api_token = self._parse_json(nebula_cookie_value, None).get('apiToken')
 
         # try to authenticate using credentials if no valid token has been found
-        if not self._nebula_token:
-            self._nebula_token = self._retrieve_nebula_auth()
+        if not self._nebula_api_token:
+            self._nebula_api_token = self._retrieve_nebula_auth()
+
+        # get a Bearer token for the Nebula API, that we need to fetch meta data
+        self._nebula_bearer_token = self._fetch_nebula_bearer_token()
+
+        # get a Zype access token which is required to access video streams (for us: to generate video URLs)
+        self._zype_access_token = self._fetch_zype_access_token()
+
+
+class NebulaIE(NebulaBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?(?:watchnebula\.com|nebula\.app)/videos/(?P<id>[-\w]+)'
+    _TESTS = [
+        {
+            'url': 'https://nebula.app/videos/that-time-disney-remade-beauty-and-the-beast',
+            'md5': 'fe79c4df8b3aa2fea98a93d027465c7e',
+            'info_dict': {
+                'id': '5c271b40b13fd613090034fd',
+                'ext': 'mp4',
+                'title': 'That Time Disney Remade Beauty and the Beast',
+                'description': 'Note: this video was originally posted on YouTube with the sponsor read included. We weren’t able to remove it without reducing video quality, so it’s presented here in its original context.',
+                'upload_date': '20180731',
+                'timestamp': 1533009600,
+                'channel': 'Lindsay Ellis',
+                'channel_id': 'lindsayellis',
+                'uploader': 'Lindsay Ellis',
+                'uploader_id': 'lindsayellis',
+            },
+            'params': {
+                'usenetrc': True,
+            },
+            'skip': 'All Nebula content requires authentication',
```

Comment:
We don't run download tests in GHA anymore. So you can remove the skip. Anyone running this test will be already aware that creds are needed